core/src/smartcontracts/isi/triggers.rs

```diff
@@ -157,8 +160,25 @@ pub mod isi {
             authority: <Account as Identifiable>::Id,
             wsv: &WorldStateView,
         ) -> Result<(), Self::Error> {
-            wsv.execute_trigger(self.trigger_id, authority);
-            Ok(())
+            let id = self.trigger_id;
+
+            wsv.triggers()
+                .inspect_by_id(&id, |action| -> Result<(), Self::Error> {
+                    let allow_execute =
+                        if let FilterBox::ExecuteTrigger(filter) = action.clone_and_box().filter {
+                            let event = ExecuteTriggerEvent::new(id.clone(), authority.clone());
+                            filter.matches(&event) || action.technical_account() == &authority
```

Comment:
It seems these checks should be in `handle_execute_trigger_event`.

Reply:
How to handle error in this case and determine what's happened: it was unauthorized or it was not found?

Reply:
Also, all implementations of execute in this module can be moved to `Set` implementation then.

Reply:
I think this problem will resolve once we moved the ISI into a separate crate.